### ISSUE TITLE
Support replacement scans

### ISSUE THREAD

**Issue by alexeyzimarev** [Giorgi/DuckDB.NET#239]
Replacement scans allow table functions to be used easier, and the API doesn't seem to complex (although a bit hard to understand). It'd be cool to get replacement scans to be supported by Duck.Net.

**Comment by Giorgi** [comment]
I had a look at the API and replacement scans look like a slimmed version of table functions. According to the documentation **The replacement scan can then insert a table function** so you would still need to register a custom table function if you want to use that function in a replacement scan.

**Comment by alexeyzimarev** [comment]
Yeah, exactly. I already have a table function, it works great. Now I am planning for the function to "pretend" that it's a table.

**Comment by RusselWebber** [comment]
This functionality would be useful for the xlDuckDb project https://github.com/RusselWebber/xlDuckDb/issues/3
